src/operators.jl

```diff
@@ -858,3 +855,23 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, vs::Vector{U})  where U <: Integer
+    vs = sort(vs, rev=true)
+    v0 = vs[end]
+    for v in vs[1:end-1]
+        for u in inneighbors(g, v)
+            if !(u in vs)
+                add_edge!(g, u, v0)
+            end
+        end
+        for u in outneighbors(g, v)
+            if !(u in vs)
```

Comment:
see above. In general, membership checking of a vector is O(n), and we'd like to avoid that wherever we can.